Core/Core/Grades/GradeListViewController.swift

```diff
@@ -132,55 +151,65 @@ public class GradeListViewController: UIViewController, ColoredNavViewProtocol {
-            updateGradingPeriod(id: enrollment?.currentGradingPeriodID)
+        if !gradingPeriodLoaded, gradingPeriodID != courseEnrollment?.currentGradingPeriodID {
+            updateGradingPeriod(id: courseEnrollment?.currentGradingPeriodID)
         }
         gradingPeriodLoaded = gradingPeriodLoaded || (courses.requested && !courses.pending)
-        gradingPeriodView.isHidden = !gradingPeriodLoaded || enrollment?.multipleGradingPeriodsEnabled != true
+        gradingPeriodView.isHidden = !gradingPeriodLoaded || courseEnrollment?.multipleGradingPeriodsEnabled == false
         gradingPeriodLabel.text = gradingPeriodID == nil && gradingPeriodLoaded
             ? NSLocalizedString("All Grading Periods", bundle: .core, comment: "")
             : gradingPeriods.first(where: { $0.id == gradingPeriodID })?.title
-        let filter = gradingPeriodID == nil
-            ? NSLocalizedString("Filter", bundle: .core, comment: "")
-            : NSLocalizedString("Clear filter", bundle: .core, comment: "")
-        filterButton.setTitle(filter, for: .normal)
 
-        totalGradeLabel.text = courses.first?.hideFinalGrades == true
-            ? NSLocalizedString("N/A", bundle: .core, comment: "")
-            : enrollment?.formattedCurrentScore(gradingPeriodID: gradingPeriodID)
+        if courses.first?.hideFinalGrades == true {
+            totalGradeLabel.text = NSLocalizedString("N/A", bundle: .core, comment: "")
+        } else if gradingPeriodID != nil {
+            totalGradeLabel.text = gradeEnrollment?.formattedCurrentScore(gradingPeriodID: gradingPeriodID)
+        } else {
+            totalGradeLabel.text = courseEnrollment?.formattedCurrentScore(gradingPeriodID: gradingPeriodID)
+        }
 
-        loadingView.isHidden = assignments.error != nil || !assignments.pending || !assignments.isEmpty || refreshControl.isRefreshing
-        emptyView.isHidden = assignments.error != nil || assignments.pending || !assignments.isEmpty
+        let isLoading = !assignments.requested || assignments.pending || !gradingPeriodLoaded
+        loadingView.isHidden = assignments.error != nil || !isLoading || !assignments.isEmpty || refreshControl.isRefreshing
+        emptyView.isHidden = assignments.error != nil || isLoading || !assignments.isEmpty
         errorView.isHidden = assignments.error == nil
         tableView.reloadData()
     }
 
     func updateGradingPeriod(id: String?) {
         gradingPeriodID = id
-        // Delete assignment groups immediately, to see a spinner again
-        assignments.useCase.reset(context: env.database.viewContext)
-        try? env.database.viewContext.save()
+        enrollments = env.subscribe(GetEnrollments(
+            context: ContextModel(.course, id: courseID),
+            userID: userID,
+            gradingPeriodID: gradingPeriodID,
+            types: [ "StudentEnrollment" ],
+            states: [ .active ]
```

Comment:
I haven't tested this yet but I suspect that you will need to `include: [.observed_users]` for parent grades.

Reply:
the previous one wasn't and the screen seems to work in Parent.